### PR TITLE
Closes #2577: Add unit tests for set_admiral_options

### LINKED ISSUE #2577: Enhance test coverage for `set_admiral_options()`
### Background Information

It would be great to get `{admiral}`'s test coverage to 100%. However, this should **not** come at the expense of development time/effort from `{admiral}` core developers. As such, I have made a set of issues tagged "Unit Tests" (such as this one for set_admiral_options()`) where users who wish to dip their feet into the `{admiral}` ecosystem can:

- Learn a bit about unit testing for R packages
- Make a meaningful contribution to {admiral}

by adding or modifying unit tests to account for lines not previously covered.

**Important note: Given the above, this issue is for newcomers to `{admiral}` or users who wish to contribute to the package, and should not be taken on by core `{admiral}` developers.**

### Definition of Done

- [ ] Unit test(s) added/modified to test the lines in red in `set_admiral_options()`:
image

### PR DESCRIPTION
**Pull request by Anthogonyst**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Umm...maybe we should re-look at the interactive portion of this function??

**Comment by manciniedoardo** [comment]
Happy to reevaluate the testing for this function, but I do think the best place for it remains the admiral package itself.

**Comment by bms63** [comment]
> roxygen_param_by_vars

We should split out from this PR/ISsue as roxygen_param_by_vars was an experimental function.  We don't really have proper documentation for it.  Fine if we want to add tests to it, but we should probably also document it now that it is being used more 
 (https://github.com/pharmaverse/admiral/blob/main/R/roxygen2.R)

File: tests/testthat/test-admiral_options.R [hunk 1]
```diff
@@ -35,4 +35,26 @@ test_that("set_admiral_options Test 5: unexpected function input for set gives e
...
+    class = "assert_integer_scalar"
+  )
+  sigfigs <- -1
+  expect_error(
+    set_admiral_options(signif_digits = sigfigs),
+    class = "assert_integer_scalar"
+  )
+})
+
+## Test 7: set works ----
```
Comment (manciniedoardo) [inline tests/testthat/test-admiral_options.R:53]: Minor comment: please update the title of Test 7 and Test 4 to be a bit more descriptive, i.e. specify which option you are testing?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @Anthogonyst look great! Just a minor comment.

**Comment by bms63** [comment]
@Anthogonyst are you able to complete this? - we split out the tests so this is the only one to complete now.

**Comment by Anthogonyst** [comment]
Sure thing. Here you go, the test comments are updated.  @manciniedoardo @bms63

**Review by bms63 — APPROVED** [review]
Thanks!!   not sure what adventure you want to go on #2605 is a quick one but kind of interesting i think

**Comment by manciniedoardo** [comment]
@Anthogonyst - totally up to you, but if you do want to pass us your name we can [acknowledge your contribution in the README](https://github.com/pharmaverse/admiral/pull/2621)

**Comment by Anthogonyst** [comment]
@bms63 @manciniedoardo  If you can add my name as "Anthony Arroyo" in the contributions list, that would be great. I'm not entirely sure what my next steps are at the moment. Thank you.

**Pull request merged**